### PR TITLE
Fixing tests

### PR DESCRIPTION
**Pull request by pstibrany**
**What this PR does**: This PR updates snappy package to latest master version (it has fix for arm64 assembly code, which panics in v0.0.2 when running on upcoming go1.16) and fixes tests to run on darwin/arm64 platform.

This also partially fixes https://github.com/cortexproject/cortex/issues/3579 (`TestWatchPrefix/etcd` test).

(Running with go1.16rc1)

**Checklist**
- [x] Tests updated
- [na] Documentation added
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ring/kv/kv_test.go [hunk 1]
```diff
@@ -197,6 +201,7 @@ func TestWatchPrefix(t *testing.T) {
 				}
 				require.NoError(t, err)
 			}
+			fmt.Println("Generator finished in", time.Since(start))
```
Comment (pracucci) [inline pkg/ring/kv/kv_test.go:204]: If you need to keep these `fmt.Println()`, what do you think about `t.Log()` instead? Otherwise I would drop them.

**Review by pracucci — APPROVED** [review]
LGTM, thanks! So, should I assume you've bought the new ARM-based MacBook? Jokes apart, is the snappy upgrade worth a CHANGELOG entry about what it fixes?

File: pkg/ring/kv/kv_test.go (hunk 1 above)
Comment (pstibrany) [reply pkg/ring/kv/kv_test.go:204]: > is the snappy upgrade worth a CHANGELOG entry about what it fixes?

It only fixes arm64 code on (still unreleased) go1.16. Given that, I don't think it's worth the changelog entry.

**Comment by pstibrany** [comment]
> is the snappy upgrade worth a CHANGELOG entry about what it fixes?

It only fixes arm64 code on (still unreleased) go1.16. Given that, I don't think it's worth the changelog entry.

**Pull request merged**